File: worksheets/dp-three/scala/src/prototype/AccessControlProvider.scala
package prototype

object AccessControlProvider {

  private var map = ???

  println("Fetching data from external resources and creating access control objects...")

  // UNCOMMENT
  //map.put("USER", new AccessControl("USER", "DO_WORK"))

  //map.put("ADMIN", new AccessControl("ADMIN", "ADD/REMOVE USERS"))

  //map.put("MANAGER", new AccessControl("MANAGER", "GENERATE/READ REPORTS"))

  //map.put("VP", new AccessControl("VP", "MODIFY REPORTS"))

  def getAccessControlObject(controlLevel: String): AccessControl = ???

}

File: worksheets/dp-three/kotlin/src/prototype/AccessControlProvider.kt
package prototype

object AccessControlProvider {

    private var map: MutableMap<String, AccessControl> = TODO()

    init {
        println("Fetching data from external resources and creating access control objects...")

        map.put("USER", AccessControl("USER", "DO_WORK"))
        map.put("ADMIN", AccessControl("ADMIN", "ADD/REMOVE USERS"))
        map.put("MANAGER", AccessControl("MANAGER", "GENERATE/READ REPORTS"))
        map.put("VP", AccessControl("VP", "MODIFY REPORTS"))
    }

    fun getAccessControlObject(controlLevel: String): AccessControl {
        TODO()
    }

}
